packages/@lwc/engine-dom/src/createFragment.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isUndefined } from '@lwc/shared';
+
+const SUPPORTS_TEMPLATE = typeof HTMLTemplateElement === 'function';
+
+export let createFragment: (html: string) => Node | null;
+
+if (SUPPORTS_TEMPLATE) {
+    // Parse the fragment HTML string into DOM
+    createFragment = function (html: string) {
+        const template = document.createElement('template');
+        template.innerHTML = html;
+        return template.content.firstChild;
+    };
```

Comment:
I considered caching and re-using the `template` here, but I decided against it because:

1. This is just a one-time setup cost anyway – it wouldn't improve any of our benchmarks
2. Holding onto the memory from the last HTML in the cached `template` strikes me as a little weird
3. I'm not even sure if it's a big perf improvement in the first place

We might explore caching the `template` in a follow-up PR, though.